HIP/hip-1056.md

```diff
@@ -0,0 +1,1787 @@
+
+A block item is a logical separation of components that together make up a block. A Block item will fall into one of 3
+categories - input, output or other. A Block item may be one of the 10 initial types noted below.
+
+```protobuf
+message BlockItem {
+    // Reserved for future items that are neither "input" nor "output".
+    reserved 11,12,13,14,15;
+
+    oneof item {
+        /**
+         * An header for the block, marking the start of a new block.
+         */
+        com.hedera.hapi.block.stream.output.BlockHeader block_header = 1;
+
+        /**
+         * An header emitted at the start of a new network "event".
+         */
+        com.hedera.hapi.block.stream.input.EventHeader event_header = 2;
+
+        /**
+         * An header emitted at the start of a new consensus "round".
+         * <p>
+         * This item SHALL contain the properties relevant to a single
+         * consensus round.
+         */
+        com.hedera.hapi.block.stream.input.RoundHeader round_header = 3;
+
+        /**
+         * A single transaction.
+         */
+        com.hedera.hapi.platform.event.EventTransaction event_transaction = 4;
+
+        /**
+         * The result of running a transaction.
+         */
+        com.hedera.hapi.block.stream.output.TransactionResult transaction_result = 5;
+
+        /**
+         * A transaction output.
+         */
+        com.hedera.hapi.block.stream.output.TransactionOutput transaction_output = 6;
+
+        /**
+         * A set of state changes.
+         */
+        com.hedera.hapi.block.stream.output.StateChanges state_changes = 7;
+
+        /**
+         * Verification data for an item filtered from the stream.<br/>
+         * This is a hash for a merkle tree node where the contents of that
+         * part of the merkle tree have been removed from this stream.
+         */
+        FilteredItemHash filtered_item_hash = 8;
+
+        /**
+         * A signed block proof.<br/>
+         * The signed merkle proof for this block. This will validate
+         * a "virtual" merkle tree containing the previous block "virtual"
+         * root, an "input" subtree, an "output" subtree, and
+         * a "state changes" subtree.
+         */
+        BlockProof block_proof = 9;
+
+        /**
+         * A record file and associated data.
+         */
+        RecordFileItem record_file = 10;
+    }
+}
+
+```
+
+All items in the block other than the final `BlockProof`, `RecordFileItem` and `FilteredItemHash` are either Input or
+Output items. This separation is important as they are hashed into two different sub-merkle-trees in the final proof.
+We did not want to carry in each block item data representing which tree it was hashed in as that would waste a lot of
+bytes. So which tree is defined _both_ by the protobuf package that message type is in and by a convention for the
+_file number_ assigned to the item in the this message. The reason they are split into input and output item sets is
```

Comment:
Correction (we have field numbers, not file numbers):
```suggestion
_field number_ assigned to the item in this message. The reason they are split into input and output item sets is
```

Reply:
Fixed